src/libraries/Directory.Build.targets

```diff
@@ -295,6 +295,8 @@
 
   <PropertyGroup>
     <BeforePack>IncludeAnalyzersInPackage;$(BeforePack)</BeforePack>
+    <BuildAnalyzerReferences>$(BuildProjectReferences)</BuildAnalyzerReferences>
+    <BuildAnalyzerReferences Condition="'$(BuildingInsideVisualStudio)' == 'true'">false</BuildAnalyzerReferences>
```

Comment:
Why aren't we building analyzers inside? Just because they are missing from the sln? I can easily fix that in the slngen configuration.

Reply:
No, we put them in the SLN as they are in the closure of the tests.  This is set to false, otherwise the VS & BuildProjectReferences will still try to call `GetTargetPath` rather than `Build` which will result in an error.  See the comment below.

Reply:
To put it another way, VS never calls "Build" across a `ProjectReference`, but it will call `GetTargetPath` unless we set this.  See https://github.com/dotnet/msbuild/issues/5434.  The call to `GetTargetPath` fails because it's calling on the outer build.